ColorPicker: Fixed height when switching tabs

DanMPA opened the pull request:
Fixed the component's height by setting a consistent height and adding margin. Also added a test to ensure the component heights remain nearly identical when switching tabs.

fixes #99675

https://github.com/user-attachments/assets/93105a1e-2084-41c7-9762-27f4150a8513

DanMPA committed:
ColorPicker: fixed height when switching tabs

Fixed the component's height by setting a consistent height and adding margin.
Also added a test to ensure the component heights remain nearly identical when switching tabs.

Fix #99675

Clarity-89 commented on packages/grafana-ui/src/components/ColorPicker/SpectrumPalette.tsx:
```diff
@@ -46,9 +46,11 @@ const SpectrumPalette = ({ color, onChange }: SpectrumPaletteProps) => {
 export const getStyles = (theme: GrafanaTheme2) => ({
   wrapper: css({
     flexGrow: 1,
+    height: '275px',
```
This component can be used in other places outside of `ColorPickerPopover`, so fixing its height is not optimal. I think the change needs to be done in the `ColorPickerPopover` instead.

DanMPA replied:
I see your point about keeping the component flexible. Should the height adjustment be applied to the popover container itself or just to its content? If the content exceeds the container’s height, do we want to allow scrolling? f not it expands dynamically and, we would still have the same issue.

DanMPA replied:
Something like this, if the content extends past the height as a fall back:

https://github.com/user-attachments/assets/a47abd11-b70a-457b-a863-c12b3ad8c48a

The height of the content was reduced to simulate the behaviour.

Clarity-89 replied:
I think the color options are pretty much set, so the height shouldn't change that much?

DanMPA committed:
ColorPickerPopover: Fixed height for popover content and allowed overflow. #99675

Clarity-89 commented on packages/grafana-ui/src/components/ColorPicker/SpectrumPalette.tsx:
```diff
@@ -49,6 +49,7 @@ export const getStyles = (theme: GrafanaTheme2) => ({
   }),
   root: css({
     '&.react-colorful': {
+      marginTop: theme.spacing(2),
```
Is this still necessary?

DanMPA replied:
It is not necessary to have the extra margin; it was just for aesthetics.

Without spacing:
<img width="300" alt="Screenshot 2025-04-02 at 2 06 51 PM" src="https://github.com/user-attachments/assets/d518c4da-5164-413e-aeac-aa1795f34d35" />

With spacing:
<img width="300" alt="Screenshot 2025-04-02 at 2 05 13 PM" src="https://github.com/user-attachments/assets/24c3e51e-89fd-4965-95ba-77bff159bd8d" />

Clarity-89 reviewed (commented):
@DanMPA could you update your branch to the latest main to (hopefully) fix the CI pipelines?

kylebrandt committed:
SQL Expressions: Make SQL Expressions work with Alerting (#101820)

Initial support for alerting with SQL expressions

- When `format` is set to `alerting`, SQL expressions output in a format suitable for alerting evaluation.
- Outstanding TODOs:
  - Deduplicate output rows
  - Add more tests
  - Fix broken alerting UI rendering (likely due to shape change to undocumented full-long format)
- Basic usage:
  - SQL must return one numeric column and one or more string columns.
  - Each row may become an alert.
  - The alert fires if the numeric value is non-zero.
  - String columns are treated as labels.

---------

Co-authored-by: Konrad Lalik <konradlalik@gmail.com>
Co-authored-by: Tom Ratcliffe <tom.ratcliffe@grafana.com>
Co-authored-by: Sam Jewell <sam.jewell@grafana.com>

torkelo committed:
DashboardScene: Add library panel via canvas add panel action  (#103254)

* Update

* It works now for auto grid as well

svennergr committed:
Investigations: add `fieldConfig` and `queries` (#103276)

* Investigations: add `fieldConfig` and `queries`

* fix openAPI

bfmatei committed:
Dynamic Dashboards: Add drag and drop for tabs and rows (#103216)

DanMPA committed:
ColorPickerPopover : Fix formatting suggested by linter.

aocenas committed:
Update codeowners for locale files (#103268)

Update codeowners

alyssajoyner committed:
Azure Monitor: Add logs query builder (#99055)

dprokop committed:
Schema V2: Use App SDK generated types (#103100)

* Use App SDK generated types

* Move examples

* Merge fix

* PAth updates

adriaanmolendijk committed:
docs: Add missing RBAC Action Definition Contact Point Receiver Test (#103273)

Update index.md

ifrost committed:
Tempo: Update lezer-traceql with new functions (#103282)

Clarity-89 commented on packages/grafana-ui/src/components/ColorPicker/ColorPickerPopover.test.tsx:
```diff
@@ -32,6 +32,23 @@ describe('ColorPickerPopover', () => {
       expect(color).toHaveStyle('box-shadow: inset 0 0 0 2px #73BF69,inset 0 0 0 4px #000000');
     });
+
+    it('should maintain the same height when switching tabs', async () => {
```
This test is very specific about the DOM order and will break once that's changed. Tbh, I don't think we need a test for the case of the picker height.

ryantxu committed:
Chore: Update yarn.lock deps (#103277)

vtorosyan committed:
Docs: Grafana SCIM provisioning overview  (#102183)

* Docs: Create a landing page for SCIM provisioning

* Fix docs warnings

* Update docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

* Update docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

* Apply suggestions from code review

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>

* Address some of the feedback and simplify few parts

* Remove repetitive wording

* Don't use possessive form, again

* Address the feedback from PR

* Docs: Manage users and teams via SCIM (#102478)

* Docs: Manage users and teams via SCIM

* Docs: Manage users and teams via SCIM

* Refactor managing users parts

* Add team sync docs

* Update _index.md

* Apply suggestions from code review

Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>

* Clarify how user identity linking is working

* Fix formatting

* Docs: Configure SCIM with Azure and Okta (#102582)

* Docs: Configure SCIM with Azure and Okta

* Apply suggestions from code review

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>
Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

* Add a warning about migration from team sync

* Remove the SAML setup instructions from SCIM docs

* Update _index.md

* Update _index.md

* Apply suggestions from code review

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>
Co-authored-by: Sarah Constant <sarahleejane@users.noreply.github.com>

* Address PR feedback

* Stop using possessive form

* Add a note mentioning that users/teams can not be manually deleted

* Update _index.md

* Docs: SCIM configuration options (#103085)

* Add a section to clarify how SCIM works with other IdPs

* Update _index.md

* Update _index.md

---------

Co-authored-by: Jack Baldry <jack.baldry@grafana.com>
Co-authored-by: linoman <2051016+linoman@users.noreply.github.com>
Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>
Co-authored-by: Sarah Constant <sarahleejane@users.noreply.github.com>

joey-grafana committed:
NodeGraph: Add node graph algorithm layout option (#102760)

* Add layout buttons

* Add config for node graph panel

* Tests

* Update test

* Updates

* Move grid button and cache nodes

* Remove limit and add warning

* Update default

renovate[bot] committed:
Update scenes to v6.6.2 (#103280)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

DanMPA commented on packages/grafana-ui/src/components/ColorPicker/ColorPickerPopover.tsx:
```diff
@@ -151,9 +151,11 @@ const getStyles = stylesFactory((theme: GrafanaTheme2) => {
       width: '246px',
       fontSize: theme.typography.bodySmall.fontSize,
       minHeight: '184px',
+      height: '290px',
       padding: theme.spacing(1),
       display: 'flex',
       flexDirection: 'column',
+      overflowY: 'auto',
```
Initially I tried this but a flicker happens because minHeight allows dynamic resizing, causing brief layout shifts when the new tab renders. Using height prevents this by enforcing a fixed size.
 
https://github.com/user-attachments/assets/b4620169-d690-4ed9-a229-5766a174f708

aangelisc committed:
Influx: Support PDC for Influx SQL (#103032)

* Support PDC for Influx SQL

* Correctly set the gRPC transport creds dial option

* Logging and refactor

* Allow URLs with no scheme

* Use passthrough resolver for socks proxy

* Update comment

* Extract url parsing and add test

* Check err

DanMPA commented:
Sorry about the mess. I did something wrong on git.

DanMPA commented:
Sorry for the mess this is my first time. Going to open a new pull request and will be more careful with git.

DanMPA commented:
@Clarity-89 Thank you for your patience. I have opened a new PR with the changes.
